src/mappings/entities/mapConfidentialAssetTransaction.ts

```diff
@@ -0,0 +1,105 @@
+import { Codec } from '@polkadot/types/types';
+import {
+  EventIdEnum,
+  ModuleIdEnum,
+  ConfidentialAssetHolder,
+  ConfidentialAssetTransaction,
+} from '../../types';
+import { getTextValue } from '../util';
+import { HandlerArgs } from './common';
+
+type AssetHolderParams = {
+  id: string;
+  accountId: string;
+  assetId: string;
+  amount: string;
+  balance: string;
+};
+
+const extractAssetHolderParams = (params: Codec[]): AssetHolderParams => {
+  const [rawAccountId, rawAssetId, rawAmount, rawBalance] = params;
+
+  const accountId = getTextValue(rawAccountId);
+  const assetId = getTextValue(rawAssetId);
+  const amount = getTextValue(rawAmount);
+  const balance = getTextValue(rawBalance);
+  const id = `${assetId}/${accountId}`;
+
+  return { id, accountId, assetId, amount, balance };
+};
+
+const handleConfidentialTransaction = async ({
+  blockId,
+  eventIdx,
+  eventId,
+  params,
+  block,
+}: HandlerArgs): Promise<void> => {
+  const { id, accountId, assetId, amount, balance } = extractAssetHolderParams(params);
+
+  const existingAssetHolder = await ConfidentialAssetHolder.get(id);
+
+  if (existingAssetHolder) {
+    existingAssetHolder.amount = balance;
+    existingAssetHolder.updatedBlockId = blockId;
+
+    await existingAssetHolder.save();
+  } else {
+    await ConfidentialAssetHolder.create({
+      id,
+      accountId,
+      assetId,
+      amount,
```

Comment:
```suggestion
      amount: balance,
```
Just for uniformity (that we update balance here), even though for the first time both will be same